### ISSUE TITLE
Does MOVETYPE_FLYMISSILE actually work?

### ISSUE THREAD

**Issue by johnstt** [ValveSoftware/halflife#1751]
This is definitely an engine issue, but there's no repo for the engine so I'll write down everything I know here! 

I noticed the only place MOVETYPE_FLYMISSILE is used in the SDK code is in DMC ([nails](https://github.com/ValveSoftware/halflife/blob/master/dmc/dlls/quake_nail.cpp#L63), [rockets](https://github.com/ValveSoftware/halflife/blob/master/dmc/dlls/quake_rocket.cpp#L44)). This seems to have been done to mimic the original Quake code for nails/rockets.

I tried putting it on some projectiles in my mod, and it did not seem to affect hit registration at all.

It did, however, make my projectiles stop having gravity and friction applied (which seems like a pretty arbitrary and weird thing to do).

If the xash3d code is to be believed, flymissile [only changes the clipsize](https://github.com/Xash3DLinux/xash3dlinux/blob/master/engine/server/sv_world.c#L1279-L1296) of moving things _after it has already collided with something_. 

In other words, it's not "// extra size to monsters" as the SDK says. Its size is whatever you had set, and then the clipsize changes after it already hit or did not hit (using the regular size to trace against monster hulls).

The xash3d code also shows that [friction and gravity are not applied](https://github.com/Xash3DLinux/xash3dlinux/blob/master/engine/server/sv_phys.c#L1396-L1413).

In Quake this code actually works because it changes the size of projectiles [whether they have already collided or not](https://github.com/id-Software/Quake/blob/master/QW/server/world.c#L842-L850). I bet it does nothing in DMC, and Quake probably has better nail/rocket registration than DMC because of it.

Welp that's all folks!

**Comment by SamVanheer** [comment]
The engine's version of this code appears to be identical to that of Xash, so your assertions are correct. My guess is they changed how it works for some reason we'll never know, but since they don't use those move types for any new stuff nobody ever noticed that anything was different.
